### PR TITLE
GCP 1.8.0 template version refresh

### PR DESCRIPTION
**Pull request by anandkarra**
Updated `gcp-compute-persistent-disk-csi-driver` storage template to version `1.8.0`.

Issue link: https://github.ibm.com/alchemy-containers/armada-storage/issues/3894

### CLI template list output

```
❯ ic sat storage template ls
Listing Satellite storage templates...
OK
Name                                     Display Name                                   Versions   
aws-ebs-csi-driver                       AWS EBS CSI Driver                             0.9.14 (deprecated), 1.1.0, 1.5.1 (default)   
aws-efs-csi-driver                       AWS EFS CSI Driver                             1.2.2 (deprecated), 1.3.1, 1.3.7 (default)   
azuredisk-csi-driver                     Azure Disk CSI Driver                          1.4.0, 1.18.0 (default), 1.23.0   
azurefile-csi-driver                     Azure File CSI Driver                          1.9.0, 1.18.0 (default), 1.22.0   
gcp-compute-persistent-disk-csi-driver   GCP Compute Persistent Disk CSI Driver         1.0.4 (default), 1.7.1, 1.8.0   
ibm-object-storage-plugin                [Beta] IBM Object Storage Plugin.              2.2 (default)   
ibm-system-storage-block-csi-driver      IBM System Storage Block CSI driver            1.4.0 (default), 1.5.0   
ibm-vpc-block-csi-driver                 [Beta] IBM VPC Block CSI driver                4.3.0 (default)   
local-volume-block                       Local Storage Operator - Block                 4.7, 4.8, 4.9 (default), 4.10   
local-volume-file                        Local Storage Operator - File                  4.7, 4.8, 4.9 (default), 4.10   
netapp-ontap-nas                         NetApp Ontap-NAS Driver                        21.04 (default), 22.04   
netapp-ontap-san                         NetApp Ontap-SAN Driver                        21.04 (default), 22.04   
netapp-trident                           NetApp Trident Operator                        21.04 (default), 22.04   
odf-local                                OpenShift Data Foundation for local devices    4.7, 4.8, 4.9 (default), 4.10   
odf-remote                               OpenShift Data Foundation for remote storage   4.7, 4.8, 4.9 (default), 4.10   
vsphere-csi-driver                       VMware CSI Driver                              2.5.1 (default)
```

### CLI template get output

```
❯ ic sat storage template get --name=gcp-compute-persistent-disk-csi-driver --version=1.8.0
Getting Satellite storage template details...
OK
                   
Name:           gcp-compute-persistent-disk-csi-driver   
Display Name:   GCP Compute Persistent Disk CSI Driver   
Version:        1.8.0   

Secret Parameters
Name                          Display Name                                     Description                                       Required   Type     Default   Mutable   
auth_provider_x509_cert_url   URL for the authorization provider certificate   URL for the authorization provider certificate.   true       string   -         true   
auth_uri                      Authorization URI                                Authorization URI for the service account.        true       string   -         true   
client_email                  Client email                                     Client email.                                     true       string   -         true   
client_id                     Client ID                                        Client ID.                                        true       string   -         true   
client_x509_cert_url          URL for the client certificate                   URL for the client certificate.                   true       string   -         true   
private_key                   Private key of the service account               Private key of the service account.               true       string   -         true   
private_key_id                Google Cloud private key ID                      Google Cloud private key ID.                      true       string   -         true   
project_id                    Google Cloud project ID                          Google Cloud project ID.                          true       string   -         true   
token_uri                     Token URI                                        Token URI for the service account.                true       string   -         true   

Storage Class Parameters
Name                       Display Name               Description                                                                              Required   Type   Default   
storage-class-preference   Storage Class Preference   Specify a unique volume tier preference; in the range of 1-99 with 1 being the highest   false      text   99   
type                       Disk type                  The disk type to use for the storage class.                                              true       -      -   
volumeBindingMode          Volume binding mode        The volume binding mode.                                                                 false      -      WaitForFirstConsumer   

Pre-defined storage class parameters
Storage Class Name             Parameter Name                                Parameter Value   
sat-gce-block-bronze-metro     is-user-defined-class                         false   
                               provisioner                                   pd.csi.storage.gke.io   
                               storage-class-preference                      4   
                               storageclass.kubernetes.io/is-default-class   false   
                               type                                          pd-standard   
                               volume-type                                   block   
                               volumeBindingMode                             WaitForFirstConsumer   
                                                                                
sat-gce-block-silver-metro     is-user-defined-class                         false   
                               provisioner                                   pd.csi.storage.gke.io   
                               storage-class-preference                      3   
                               storageclass.kubernetes.io/is-default-class   false   
                               type                                          pd-balanced   
                               volume-type                                   block   
                               volumeBindingMode                             WaitForFirstConsumer   
                                                                                
sat-gce-block-gold-metro       is-user-defined-class                         false   
                               provisioner                                   pd.csi.storage.gke.io   
                               storage-class-preference                      2   
                               storageclass.kubernetes.io/is-default-class   true   
                               type                                          pd-ssd   
                               volume-type                                   block   
                               volumeBindingMode                             WaitForFirstConsumer   
                                                                                
sat-gce-block-platinum-metro   is-user-defined-class                         false   
                               provisioner                                   pd.csi.storage.gke.io   
                               storage-class-preference                      1   
                               storageclass.kubernetes.io/is-default-class   false   
                               type                                          pd-extreme   
                               volume-type                                   block   
                               volumeBindingMode                             WaitForFirstConsumer   
                                                                                
sat-gce-block-bronze           is-user-defined-class                         false   
                               provisioner                                   pd.csi.storage.gke.io   
                               storage-class-preference                      8   
                               storageclass.kubernetes.io/is-default-class   false   
                               type                                          pd-standard   
                               volume-type                                   block   
                               volumeBindingMode                             Immediate   
                                                                                
sat-gce-block-silver           is-user-defined-class                         false   
                               provisioner                                   pd.csi.storage.gke.io   
                               storage-class-preference                      7   
                               storageclass.kubernetes.io/is-default-class   false   
                               type                                          pd-balanced   
                               volume-type                                   block   
                               volumeBindingMode                             Immediate   
                                                                                
sat-gce-block-gold             is-user-defined-class                         false   
                               provisioner                                   pd.csi.storage.gke.io   
                               storage-class-preference                      6   
                               storageclass.kubernetes.io/is-default-class   false   
                               type                                          pd-ssd   
                               volume-type                                   block   
                               volumeBindingMode                             Immediate   
                                                                                
sat-gce-block-platinum         is-user-defined-class                         false   
                               provisioner                                   pd.csi.storage.gke.io   
                               storage-class-preference                      5   
                               storageclass.kubernetes.io/is-default-class   false   
                               type                                          pd-extreme   
                               volume-type                                   block   
                               volumeBindingMode                             Immediate
```

### Create storage config

```
❯ ibmcloud sat storage config create --name ak-gcp180-20221103-1407 --template-name gcp-compute-persistent-disk-csi-driver --template-version 1.8.0 --location stage-e2e-gcp-loc  -p 'project_id=$PROJECT_ID' -p 'private_key_id=$PRIVATE_KEY_ID' -p 'private_key=$PRIVATE_KEY' -p 'client_email=$CLIENT_EMAIL' -p 'client_id=$CLIENT_ID' -p 'auth_uri=$AUTH_URI' -p 'token_uri=$TOKEN_URI' -p 'auth_provider_x509_cert_url=$AUTH_PROVIDER_X509_CERT_URL' -p 'client_x509_cert_url=$CLIENT_X509_CERT_URL'
Creating Satellite storage configuration...
OK
Storage configuration 'ak-gcp180-20221103-1407' was successfully created with ID 'd35fe267-6d6f-4ba9-8206-f4552a862909'.
```

```
❯ ic sat storage config get --config=ak-gcp180-20221103-1407
Getting Satellite storage configuration details...
OK

Name:                         ak-gcp180-20221103-1407   
ID:                           d35fe267-6d6f-4ba9-8206-f4552a862909   
Created:                      2022-11-03T08:43:18.614Z   
Template Name:                gcp-compute-persistent-disk-csi-driver   
Template Version:             1.8.0   
Template Version:             supported   
Location:                     stage-e2e-gcp-loc   
Current Revision:             1   
Latest Revision:              1   
Revision Upgrade Available:   false   

Secret Parameters
Name                          Value   
auth_provider_x509_cert_url   *****   
auth_uri                      *****   
client_email                  *****   
client_id                     *****   
client_x509_cert_url          *****   
private_key                   *****   
private_key_id                *****   
project_id                    *****   
token_uri                     *****   

Storage Class Parameters
Storage Class Name             Parameter Name             Parameter Value   
sat-gce-block-bronze-metro     is-default-class           false   
                               storage-class-preference   4   
                               type                       pd-standard   
                               volumeBindingMode          WaitForFirstConsumer   

sat-gce-block-silver-metro     is-default-class           false   
                               storage-class-preference   3   
                               type                       pd-balanced   
                               volumeBindingMode          WaitForFirstConsumer   

sat-gce-block-gold-metro       is-default-class           true   
                               storage-class-preference   2   
                               type                       pd-ssd   
                               volumeBindingMode          WaitForFirstConsumer   

sat-gce-block-platinum-metro   is-default-class           false   
                               storage-class-preference   1   
                               type                       pd-extreme   
                               volumeBindingMode          WaitForFirstConsumer   

sat-gce-block-bronze           is-default-class           false   
                               storage-class-preference   8   
                               type                       pd-standard   
                               volumeBindingMode          Immediate   

sat-gce-block-silver           is-default-class           false   
                               storage-class-preference   7   
                               type                       pd-balanced   
                               volumeBindingMode          Immediate   

sat-gce-block-gold             is-default-class           false   
                               storage-class-preference   6   
                               type                       pd-ssd   
                               volumeBindingMode          Immediate   

sat-gce-block-platinum         is-default-class           false   
                               storage-class-preference   5   
                               type                       pd-extreme   
                               volumeBindingMode          Immediate
```

### Create storage assignment

```
❯ ic sat storage assignment create --name=ak-gcp180-20221103-1407-assignment-1413 --cluster=cdcddqf200vjkm6ec1f0 --config=ak-gcp180-20221103-1407
Creating assignment...
OK
Assignment ak-gcp180-20221103-1407-assignment-1413 was successfully created with ID bdc73432-1964-42c3-9eee-a26c719f1f9d.
```

```
❯ ibmcloud sat storage assignment get --assignment=bdc73432-1964-42c3-9eee-a26c719f1f9d
OK
Assignment                         
Name:                           ak-gcp180-20221103-1407-assignment-1413   
ID:                             bdc73432-1964-42c3-9eee-a26c719f1f9d   
Owner:                          Anand.Karra@ibm.com   
Cluster Groups:                 -   
Cluster ID:                     cdcddqf200vjkm6ec1f0   
Cluster Name:                   stage-e2e-gcp   
Configuration:                  ak-gcp180-20221103-1407   
Configuration ID:               d35fe267-6d6f-4ba9-8206-f4552a862909   
Version:                        ak-gcp180-20221103-1407_v01   
Version ID:                     8b1e056b-ae53-4f11-ac8e-5285feaf5b52   
Assignment Type:                ChannelSubscription   
Created:                        2022-11-03T08:44:08.558Z   
Rollout Success Count:          2   
Rollout Error Count:            0   
Assignment Upgrade Available:   false
```

### Check resources deplyed on cluster

```
❯ ibmcloud ks cluster config --admin -c cdcddqf200vjkm6ec1f0
OK
The configuration for cdcddqf200vjkm6ec1f0 was downloaded successfully.

Added context for cdcddqf200vjkm6ec1f0 to the current kubeconfig file.
You can now execute 'kubectl' commands against your cluster. For example, run 'kubectl get nodes'.
If you are accessing the cluster for the first time, 'kubectl' commands might fail for a few seconds while RBAC synchronizes.
```

```
❯ oc get rr -n razeedeploy --insecure-skip-tls-verify
NAME                                                       AGE
clustersubscription-90a7f9c4-e3a4-4556-a5e0-516aba60d77b   20h
clustersubscription-bdc73432-1964-42c3-9eee-a26c719f1f9d   2m20s
clustersubscription-system-primaryorgkey                   8d
```

```
❯ kubectl get mustachetemplates -n razeedeploy --insecure-skip-tls-verify              
NAME                        AGE
ak-gcp180-20221103-1407     2m39s
test-vpc-driver-config430   20h
```

```
❯ kubectl get pods -n gce-pd-csi-driver --insecure-skip-tls-verify
NAME                                     READY   STATUS    RESTARTS   AGE
csi-gce-pd-controller-66fbc7dc4b-4x5p9   5/5     Running   0          3m11s
csi-gce-pd-node-c7g6g                    2/2     Running   0          3m11s
csi-gce-pd-node-l8sh6                    2/2     Running   0          3m11s
csi-gce-pd-node-z47cw                    2/2     Running   0          3m11s
```

```
❯ kubectl get sc --insecure-skip-tls-verify                                   
NAME                                          PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
ibmc-vpc-block-metro-5iops-tier               vpc.block.csi.ibm.io    Delete          WaitForFirstConsumer   true                   19h
ibmc-vpc-block-metro-custom                   vpc.block.csi.ibm.io    Delete          WaitForFirstConsumer   true                   19h
ibmc-vpc-block-metro-general-purpose          vpc.block.csi.ibm.io    Delete          WaitForFirstConsumer   true                   19h
ibmc-vpc-block-metro-retain-10iops-tier       vpc.block.csi.ibm.io    Retain          WaitForFirstConsumer   true                   19h
ibmc-vpc-block-metro-retain-5iops-tier        vpc.block.csi.ibm.io    Retain          WaitForFirstConsumer   true                   19h
ibmc-vpc-block-metro-retain-custom            vpc.block.csi.ibm.io    Retain          WaitForFirstConsumer   true                   19h
ibmc-vpc-block-metro-retain-general-purpose   vpc.block.csi.ibm.io    Retain          WaitForFirstConsumer   true                   19h
sat-gce-block-bronze                          pd.csi.storage.gke.io   Delete          Immediate              false                  3m21s
sat-gce-block-bronze-metro                    pd.csi.storage.gke.io   Delete          WaitForFirstConsumer   false                  3m22s
sat-gce-block-gold                            pd.csi.storage.gke.io   Delete          Immediate              false                  3m20s
sat-gce-block-gold-metro (default)            pd.csi.storage.gke.io   Delete          WaitForFirstConsumer   false                  3m22s
sat-gce-block-platinum                        pd.csi.storage.gke.io   Delete          Immediate              false                  3m20s
sat-gce-block-platinum-metro                  pd.csi.storage.gke.io   Delete          WaitForFirstConsumer   false                  3m21s
sat-gce-block-silver                          pd.csi.storage.gke.io   Delete          Immediate              false                  3m21s
sat-gce-block-silver-metro                    pd.csi.storage.gke.io   Delete          WaitForFirstConsumer   false                  3m22s
sat-vpc-block-gold-metro (default)            vpc.block.csi.ibm.io    Delete          WaitForFirstConsumer   true                   19h
```

### PVC and pod creation

```
❯ cat gcp-pvc.yaml 
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: podpvc
spec:
  accessModes:
    - ReadWriteOnce
  storageClassName: sat-gce-block-gold-metro
  resources:
    requests:
      storage: 200Gi
```

```
❯ kubectl create -f gcp-pvc.yaml                                                                                                                                                        
persistentvolumeclaim/podpvc created
```

```
❯ cat gcp-pod.yaml 
apiVersion: v1
kind: Pod
metadata:
  name: web-server
spec:
  containers:
   - name: web-server
     image: nginx
     command:
        - "/bin/sh"
        - "-c"
        - while true; do echo $(date) >> /mnt/googledisk/outfile; sleep 1; done
     volumeMounts:
       - mountPath: /mnt/googledisk
         name: mypvc
  volumes:
   - name: mypvc
     persistentVolumeClaim:
       claimName: podpvc
       readOnly: false
```

```
❯ kubectl create -f gcp-pod.yaml
pod/web-server created
```

```
❯ kubectl get pods --insecure-skip-tls-verify      
NAME         READY   STATUS    RESTARTS   AGE
web-server   1/1     Running   0          39s
```

```
❯ kubectl get pvc --insecure-skip-tls-verify          
NAME     STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS               AGE
podpvc   Bound    pvc-5b6c40c5-0825-4087-86ff-00acf7844d77   200Gi      RWO            sat-gce-block-gold-metro   74s
```

### Verify IO operations

```
❯ kubectl exec web-server --stdin --tty -- /bin/bash                           
root@web-server:/# cat /mnt/googledisk/outfile
Thu Nov 3 08:50:45 UTC 2022
Thu Nov 3 08:50:46 UTC 2022
Thu Nov 3 08:50:47 UTC 2022
Thu Nov 3 08:50:48 UTC 2022
Thu Nov 3 08:50:49 UTC 2022
.
.
.
```

```
root@web-server:/# df -h
Filesystem      Size  Used Avail Use% Mounted on
overlay         100G   26G   75G  26% /
tmpfs            64M     0   64M   0% /dev
tmpfs           7.8G     0  7.8G   0% /sys/fs/cgroup
shm              64M     0   64M   0% /dev/shm
/dev/sda2       100G   26G   75G  26% /etc/hosts
/dev/sdb        196G   61M  196G   1% /mnt/googledisk
tmpfs            13G   24K   13G   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs           7.8G     0  7.8G   0% /proc/acpi
tmpfs           7.8G     0  7.8G   0% /proc/scsi
tmpfs           7.8G     0  7.8G   0% /sys/firmware
```

```
root@web-server:/# mount
overlay on / type overlay (rw,relatime,context="system_u:object_r:container_file_t:s0:c51,c611",lowerdir=/var/data/criorootstorage/overlay/l/UAZIA2RKG26ZHJTWYYXP4YFJV7:/var/data/criorootstorage/overlay/l/FTFIYRW7KDSXE77OYNC52MVSEB:/var/data/criorootstorage/overlay/l/I5LDLZSCYSLJZO2QCE5VCCEVEY:/var/data/criorootstorage/overlay/l/4GZONS3AQAMNGLDVM6KVMOQJNX:/var/data/criorootstorage/overlay/l/BT2C4PZA3FFBFELAOUDFO5SSWH:/var/data/criorootstorage/overlay/l/UKQX773U3VOJWEGT77ETT7FFFG,upperdir=/var/data/criorootstorage/overlay/07c3484a24126500b865f79f42ab1d766b73951afa7ba1e0113de50db0baf0d9/diff,workdir=/var/data/criorootstorage/overlay/07c3484a24126500b865f79f42ab1d766b73951afa7ba1e0113de50db0baf0d9/work)
.
.
.
/dev/sdb on /mnt/googledisk type ext4 (rw,relatime,seclabel,data=ordered)
.
.
.
```

### Verify cleanup

```
❯ kubectl delete pod web-server --insecure-skip-tls-verify
pod "web-server" deleted
```

```
❯ kubectl get pods --insecure-skip-tls-verify             
No resources found in default namespace.
```

```
❯ cat gcp-pod.yaml 
apiVersion: v1
kind: Pod
metadata:
  name: web-server-2
spec:
  containers:
   - name: web-server-2
     image: nginx
     command:
        - "/bin/sh"
        - "-c"
        - while true; do echo $(date) >> /mnt/googledisk/outfile; sleep 1; done
     volumeMounts:
       - mountPath: /mnt/googledisk
         name: mypvc
  volumes:
   - name: mypvc
     persistentVolumeClaim:
       claimName: podpvc
       readOnly: false
```

```
❯ kubectl apply -f gcp-pod.yaml 
pod/web-server-2 created
```

```
❯ kubectl get pods --insecure-skip-tls-verify           
NAME           READY   STATUS    RESTARTS   AGE
web-server-2   1/1     Running   0          69s
```

```
❯ kubectl get pvc --insecure-skip-tls-verify            
NAME     STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS               AGE
podpvc   Bound    pvc-5b6c40c5-0825-4087-86ff-00acf7844d77   200Gi      RWO            sat-gce-block-gold-metro   9m44s
```

```
❯ kubectl exec web-server-2 --stdin --tty -- /bin/bash                         
root@web-server-2:/# cat /mnt/googledisk/outfile
Thu Nov 3 08:50:45 UTC 2022
Thu Nov 3 08:50:46 UTC 2022
Thu Nov 3 08:50:47 UTC 2022
Thu Nov 3 08:50:48 UTC 2022
Thu Nov 3 08:50:49 UTC 2022
.
.
.
```

### Cleanup resources

```
❯ kubectl delete pod web-server-2 --insecure-skip-tls-verify
pod "web-server-2" deleted
```

```
❯ kubectl delete pvc podpvc --insecure-skip-tls-verify
persistentvolumeclaim "podpvc" deleted
```

```
❯ kubectl get pods --insecure-skip-tls-verify
No resources found in default namespace.
```

```
❯ kubectl get pvc --insecure-skip-tls-verify
No resources found in default namespace.
```

### CODE REVIEW

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/storage-class-template.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: "{{ name }}"
+  labels:
+    app.kubernetes.io/name: gcp-compute-persistent-disk-csi-driver
+    app.kubernetes.io/version: 1.8.0
+    volume-type: "block"
+    storage-class-preference: "{{ storage-class-preference }}"
+    is-user-defined-class: "true"
```
Comment (gauravAggrahari) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/storage-class-template.yaml:12]: Please add this 
```
annotations:
    storageclass.kubernetes.io/is-default-class: "{{ is-default-class }}"
```

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/storage-class-parameters.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        "templateKeyName": [
+            "storage-class-preference"
+        ],
+        "required": "false",
+        "default": "99",
+        "category": "config",
+        "regex": "^[1-9]([0-9]?)$",
+        "type": "text"
+    }
+]
```
Comment (gauravAggrahari) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/storage-class-parameters.json:58]: Please add this
```
{
        "description": "Specify 'true' or 'false' to make the created storage class the default class.",
        "displayname": "Default Storage Class",
        "name": "is-default-class",
        "templateKeyName": [
            "storageclass.kubernetes.io/is-default-class"
        ],
        "default": "false",
        "required": "false",
        "category": "config",
        "type": "boolean"
    }
```

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,848 @@
...
+            app.kubernetes.io/name: gcp-compute-persistent-disk-csi-driver
+            app.kubernetes.io/version: 1.8.0
+          annotations:
+            autoUpdate: "AUTO-UPDATE-ENABLED"
+        spec:
+          containers:
+          - args:
+            - --v=5
+            - --csi-address=/csi/csi.sock
+            - --kubelet-registration-path=/var/lib/kubelet/plugins/pd.csi.storage.gke.io/csi.sock
```
Comment (gauravAggrahari) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml]: I think here also we need to change the path to /var/data

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,848 @@
...
+          - args:
+            - --v=5
+            - --endpoint=unix:/csi/csi.sock
+            - --run-controller-service=false
+            image: k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver:v1.7.2
+            name: gce-pd-driver
+            securityContext:
+              privileged: true
+            volumeMounts:
+            - mountPath: /var/lib/kubelet
```
Comment (gauravAggrahari) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml]: change this to /var/data/

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,848 @@
...
+          hostNetwork: true
+          nodeSelector:
+            kubernetes.io/os: linux
+          priorityClassName: csi-gce-pd-node
+          serviceAccountName: csi-gce-pd-node-sa
+          tolerations:
+          - operator: Exists
+          volumes:
+          - hostPath:
+              path: /var/lib/kubelet/plugins_registry/
```
Comment (gauravAggrahari) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml]: change this to /var/data/

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,848 @@
...
+              type: Directory
+            name: registration-dir
+          - hostPath:
+              path: /var/lib/kubelet
+              type: Directory
+            name: kubelet-dir
+          - hostPath:
+              path: /var/lib/kubelet/plugins/pd.csi.storage.gke.io/
```
Comment (gauravAggrahari) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml]: change this to /var/data

**Comment by gauravAggrahari** [comment]
Please paste the output for PVC creation, mount and IO on the volume.

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            "name": "provider-logo",
+            "value": "logo/gcp-compute-persistent-disk-csi-driver.svg"
+        },
+        {
+            "name": "supported-envs",
+            "value": "Google"
+        },
+        {
+            "name": "supported-k8s-versions",
+            "value": "1.20, 1.21, 1.22, 1.23, 1.24"
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: Update supported k8s versions `"1.22, 1.23, 1.24, 1.25"`

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/storage-class-parameters.json (hunk 2 above)
Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/storage-class-parameters.json:58]: Added here: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/4a19a141fdc12733b26b78e0a2fe0713410875fd

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml (hunk 6 above)
Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml]: Changed here: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/0f41c691643168e11046fa58c0916fd334695a3d

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml (hunk 5 above)
Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml]: Changed here: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/0f41c691643168e11046fa58c0916fd334695a3d

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml (hunk 4 above)
Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml]: Changed here: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/0f41c691643168e11046fa58c0916fd334695a3d

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml (hunk 3 above)
Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml]: Changed here: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/0f41c691643168e11046fa58c0916fd334695a3d

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/storage-class-template.yaml (hunk 1 above)
Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/storage-class-template.yaml:12]: Added here: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/4a19a141fdc12733b26b78e0a2fe0713410875fd

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json (hunk 7 above)
Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: Updated here: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/bc9196a5325c09cd6071a8d1426f70e9ef396706

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json [hunk 8]
```diff
@@ -0,0 +1,73 @@
+{
+    "name": "1.8.0",
+    "revision": "1",
+    "enabled": "true",
+    "status": "supported",
+    "description": "GCP Compute Persistent Disk CSI Driver",
+    "tags": [
+        {
+            "name": "provider",
+            "value": "google"
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: `google` ---> `Google`

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+            "name": "supported-envs",
+            "value": "Google"
+        },
+        {
+            "name": "supported-k8s-versions",
+            "value": "1.22, 1.23, 1.24, 1.25"
+        },
+        {
+            "name": "supported-ocp-versions",
+            "value": "4.9, 4.10, 4.11"
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: is this template  supported on 4.7 & 4.8 as well @gauravAggrahari ?
```
OpenShift Versions   
4.7.59_openshift (deprecated, unsupported in 37 days)   
4.8.51_openshift   
4.9.49_openshift   
4.10.36_openshift (default)   
4.11.8_openshift   
```

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json (hunk 8 above)
Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: Corrected: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/4c41e84e2e2e05f1c2e21cf0f58b79c927ba4fe7

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json (hunk 9 above)
Comment (gauravAggrahari) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: @anandkarra , 
We can remove this check for now.

Comment (anandkarra) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: Removed here: https://github.com/IBM/ibm-satellite-storage/pull/221/commits/0136711b3cf6db58933fa058758dbd71e248510b

**Review by mssachan — APPROVED** [review]
LGTM

**Pull request merged**